### DISCUSSION TITLE
Using EEPROM with CCButtonLatched

### DISCUSSION THREAD

**Discussion by Vahen1981** [tttapa/Control-Surface#656] [Q&A]
Hi Pieter,
Is it possible to use the EEPROM of ESP32 to save CC values?? So then I can program this values directly from the controller.
I did this before in an arduino promicro, but in ESP32 as the library is a bit different I cannot do it... 

I tried the code bellow, but is always sending CC=0
Any suggestion?

My code:

```
#include <Control_Surface.h>
#include <EEPROM.h>

HardwareSerialMIDI_Interface serialmidi = {Serial, MIDI_BAUD};
uint8_t CC1;
uint8_t CC2;
uint8_t CC3;

CCButtonLatched ComandosCC[] = {
  {25, {CC1, CHANNEL_1}},
  {32, {CC2, CHANNEL_1}},
  {23, {CC3, CHANNEL_1}},
};

void setup() {
  EEPROM.begin(8);
  eepromWrite();
  CC1 = EEPROM.read(0);
  CC2 = EEPROM.read(1);
  CC3 = EEPROM.read(2);
  Control_Surface.begin();
}

void eepromWrite(){
  EEPROM.write(0, 51);
  EEPROM.write(1, 61);
  EEPROM.write(2, 71);
}


void loop() {
  Control_Surface.loop();

}
```

thanks!

**Comment by tttapa** [comment]
The code you posted cannot work because you initialize the buttons in `ComandosCC` with a copy of the values of `CC1`-`CC3`. Changing the values of `CC1`-`CC3` later has no effect on `ComandosCC` whatsoever.

See these comments for solutions:
- https://github.com/tttapa/Control-Surface/issues/129#issuecomment-592942311
- https://github.com/tttapa/Control-Surface/issues/129#issuecomment-607512611

**Reply by Vahen1981** [reply]
Thank you Pieter, I fixed it!
